src/server/scripts/Northrend/Naxxramas/boss_noth.cpp

```diff
@@ -130,10 +174,62 @@ class boss_noth : public CreatureScript
             Talk(SAY_DEATH);
         }
 
-        void SummonUndead(uint32 entry, uint32 num)
+        void DamageTaken(Unit* /*who*/, uint32& damage) // prevent noth from somehow dying in the balcony phase
         {
-            for (uint32 i = 0; i < num; ++i)
-                me->SummonCreature(entry, SummonPos[rand32() % MAX_SUMMON_POS], TEMPSUMMON_CORPSE_DESPAWN, 60000);
+            if (!events.IsInPhase(PHASE_BALCONY))
+                return;
+            if (damage < me->GetHealth())
+                return;
+            
+            me->SetHealth(1u);
+            damage = 0u;
+        }
+
+        void CastSummon(uint8 nWarrior, uint8 nChampion, uint8 nGuardian)
+        { // this ensures we do not spawn two mobs using the same spell (<=> in the same position) if we can help it
+            uint8 it, selected;
+            uint32 temp;
+            while(nWarrior)
+                for (it = 0; it < N_WARRIOR_SPELLS && nWarrior; ++it)
+                {
+                    nWarrior--;
+                    selected = urand(it, N_WARRIOR_SPELLS - 1);
+                    DoCast(SummonWarriorSpells[selected]);
+                    if (selected != it)
+                    { // shuffle the selected into the part of the array that is no longer being searched
+                        temp = SummonWarriorSpells[selected];
+                        SummonWarriorSpells[selected] = SummonWarriorSpells[it];
+                        SummonWarriorSpells[it] = temp;
```

Comment:
it is not a good idea to modify global variables which are shared between all instances

(btw. there is a std::swap)